### ISSUE TITLE
fullpageApi is undefined

### ISSUE THREAD

**Issue by mvinet** [alvarotrigo/react-fullpage#27]
Hello,
I want to send fullpageApi in a custom React.Component but it's always undefined 

```javascript
               <ReactFullPage loopHorizontal={false}
                    render = {({ state, fullpageApi }) => {
                        return (
                            <div>
                                <MySection state={state} fullpageApi={fullpageApi} />
                            </div>
                        )
                    }}
                />
```

when I do console.log(this.props) in componentDidMount() it's return undefined for fullpageApi ..

It's normal or it's an issue ?

**Comment by cmswalker** [comment]
Thanks for reporting. This is a critical issue, seems the wrapper isn't getting diffs in render functions. I will fix soon. If you'd like to look at an example of what's going wrong I've created https://codesandbox.io/s/myl7l3rpn8

**Comment by mvinet** [comment]
any idea when it's will fix ?

**Comment by cmswalker** [comment]
@mvinet all good with the latest update, I updated the codesandbox as well https://codesandbox.io/s/myl7l3rpn8

You actually no longer have to track state.initialized yourself too, the render prop will only fire when the component is initialized

thanks for finding, this was a critical bug

**Comment by alvarotrigo** [comment]
Can we give a second though to this issue?

This is the reason why we started using a dummy section and 2 initialisations instead of one right? As the dummy section is the cause of a few issues, I believe we should look into another solution.

Isn't there any other way to deal with this issue rather than modifying the fullPage.js library and making the API available even before initialising fullpage.js?
I guess you would spect a static API in that case, which will make `fullpage_api` available by just loading the `fullpage.js` file. No initialisation required. 

Just double checking in case there's any other solution that won't involve modifying the library itself (as some developers might be checking for "fullpage_api" to check if fullPage.js was initialised, which, if we make the change, will create some issues for them as they'll need to update their code)

**Comment by alvarotrigo** [comment]
@mvinet  was the variable `state` also undefined in your first scenario?
If not, if it was only about the fullpageApi variable. Why not using the fullpage.js global variable instead? `fullpage_api`?

**Comment by cmswalker** [comment]
the global is undefined until fullpage.js initializes, which is why it's handed back in the render prop. If someone used the global + the render prop one there'd be 2 sources of truth and it could have confusing side effects

**Comment by alvarotrigo** [comment]
Fixed now in 0.1.3
